src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/mappers/AmendmentAddressToEdifactMapper.java

```diff
@@ -0,0 +1,135 @@
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AmendmentAddressToEdifactMapper extends AmendmentToEdifactMapper {
+
+    @Override
+    Segment mapPatches(AmendmentBody amendmentBody) {
+        var patches = amendmentBody.getJsonPatches();
+
+        var houseName = patches.getHouseName()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+        var numberOrRoadName = patches.getNumberOrRoadName()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+        var locality = patches.getLocality()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+        var postalTown = patches.getPostTown()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+        var county = patches.getCounty()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+        var postalCode = patches.getPostalCode()
+            .map(AmendmentPatch::getNullableFormattedSimpleValue)
+            .orElse(null);
+
+        return PersonAddress.builder()
+            .addressLine1(houseName)
+            .addressLine2(numberOrRoadName)
+            .addressLine3(locality)
+            .addressLine4(postalTown)
+            .addressLine5(county)
+            .postalCode(postalCode)
+            .build();
+    }
+
+    @Override
+    boolean shouldCreateSegment(AmendmentBody amendmentBody) {
+        var patches = amendmentBody.getJsonPatches();
+        return Stream.of(
+            patches.getHouseName(),
+            patches.getNumberOrRoadName(),
+            patches.getLocality(),
+            patches.getPostTown(),
+            patches.getCounty(),
+            patches.getPostalCode())
+            .anyMatch(Optional::isPresent);
+    }
+
+    @Override
+    protected void validatePatches(JsonPatches patches) {
+        checkIfThereAreAllFiveAddressLinesPatches(patches);
+        validateNonEmptyValues(List.of(
+            patches.getHouseName(),
+            patches.getNumberOrRoadName(),
+            patches.getLocality(),
+            patches.getPostTown(),
+            patches.getCounty(),
+            patches.getPostalCode()));
+        checkNoPostTownPatchForRemoveOperation(patches);
+        checkLocalityPostTownAndCountyAllEmptyBlankOrAllNotEmptyBlank(patches);
+    }
+
+    private void checkIfThereAreAllFiveAddressLinesPatches(JsonPatches patches) {
+        if(!Stream.of(
+            patches.getHouseName(),
+            patches.getNumberOrRoadName(),
+            patches.getLocality(),
+            patches.getPostTown(),
+            patches.getCounty())
```

Comment:
`patches.getPostalCode` is not mandatory?

Reply:
Postal code is treated different (as different data field) than actual address